### PR TITLE
added scene flow extension to the API

### PR DESCRIPTION
**Pull request by nchodosh**
## PR Summary
Updates in two places:
1. dataloaders/scene_flow.py and utils.py: Added a Flow structure and modified the data loader to return it along with the sweep pair. Also modified the behavior of the data loader so that it only iterates over valid pairs instead of only sometimes returning two sweeps.

## Testing
I've run the whole pipeline for creating flow annotations, making example dummy outputs, and running them through the evaluation code. I've also visually checked the computed flow labels and segmentation masks. Some guidance on unit tests would be appreciated.

In order to ensure this PR works as intended, it is:

* [ ] unit tested.
* [x] other or not applicable (*additional detail/rationale required*)

## Compliance with Standards

As the author, I certify that this PR conforms to the following standards:
I do not certify this
* [ ] Code changes conform to [PEP8](https://www.python.org/dev/peps/pep-0008) and docstrings conform to the Google Python [style guide](https://google.github.io/styleguide/pyguide.html?showone=Comments#38-comments-and-docstrings).
* [ ] A well-written summary explains what was done and why it was done.
* [ ] The PR is adequately tested and the testing details and links to external results are included.

### CODE REVIEW

File: src/av2/torch/dataloaders/utils.py [hunk 1]
```diff
@@ -127,6 +189,74 @@ def frame_to_tensor(frame: pd.DataFrame) -> Tensor:
...
+
+        return cls(
+            flow=torch.FloatTensor(flow),
+            valid=torch.BoolTensor(valid),
+            classes=torch.ByteTensor(classes),
+            dynamic=torch.BoolTensor(dynamic),
+        )
+
+
 def frame_to_SE3(frame: pd.DataFrame) -> Se3:
```
[the remark this answers is not in the document]
Comment (nchodosh) [reply src/av2/torch/dataloaders/utils.py:260]: This actually doesn't stop the se3 transform from getting created with requires_grad=True.

File: src/av2/evaluation/scene_flow/eval.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    Args:
+        dts: (N,3) Array containing predicted flows.
+        gts: (N,3) Array containing ground truth flows.
+
+    Returns:
+        The pointwise angle errors
+    """
+    unit_label = gts / (np.linalg.norm(gts, axis=-1, keepdims=True) + EPS)
+    unit_dts = dts / (np.linalg.norm(dts, axis=-1, keepdims=True) + EPS)
+    dot_product = np.clip(np.sum(unit_label * unit_dts, axis=-1), a_min=-1 + EPS, a_max=1 - EPS)
```
[the remark this answers is not in the document]
Comment (nchodosh) [reply src/av2/evaluation/scene_flow/eval.py:92]: This is some old code that's used in a lot of repo's for computing the angle error. I assume its there because floating point errors can make the results outside the range [-1, 1]
